File: Source/CSharp/Expressive/Expressive/Exceptions/ExpressiveException.cs
using System;

namespace Expressive.Exceptions
{
    /// <summary>
    /// The main exposed <see cref="Exception"/> for users of an Expression. Check the InnerException for more information.
    /// </summary>
#if !NETSTANDARD1_4
    [Serializable]
#endif
    public sealed class ExpressiveException : Exception
    {
        internal ExpressiveException(string message) : base(message)
        {

        }

        internal ExpressiveException(Exception innerException) : base(innerException.Message, innerException)
        {

        }
    }
}

File: Source/Java/Expressive/src/main/java/com/bijington/expressive/exceptions/ExpressiveException.java
package com.bijington.expressive.exceptions;

/**
 * Created by shaun on 11/11/2016.
 */
public class ExpressiveException extends Exception {
    /*public ExpressiveException(String message) : base(message)
    {

    }

    public ExpressiveException(Exception innerException) : base(innerException.Message, innerException)
    {

    }*/
    public ExpressiveException(String message) {
        super(message);
    }

    public ExpressiveException(Exception ex) {
        super(ex.getMessage(), ex);
    }
}
